### ISSUE TITLE
Binary keys

### ISSUE THREAD

**Issue by doublex** [Softmotions/iowow#30]
Is it possible to store binary-keys in the database?

This testcase fails (rc == IWKV_ERROR_KEY_EXISTS):
````
IWKV_OPTS opts = {
    .path = "/tmp/test.db",
    .oflags = IWKV_TRUNC
};
IWKV kv = NULL;
iwrc rc = iwkv_open( &opts, &kv );
assert( rc == 0 );
IWDB db = NULL;
rc = iwkv_db( kv, 1, 0, &db );
assert( rc == 0 );

// write (works)
{
    unsigned char ip1[4] = { 1, 0, 142, 235 };
    IWKV_val ikey, ival;
    ikey.data = ip1;
    ikey.size = 4;
    ival.data = (void*)"";
    ival.size = 0;
    iwkv_opflags opflags = IWKV_NO_OVERWRITE;
    iwrc rc = iwkv_put( db, &ikey, &ival, opflags );
    assert( rc == 0 );
}

// write (fails)
{
    unsigned char ip2[4] = { 1, 0, 145, 2 };
    IWKV_val ikey, ival;
    ikey.data = ip2;
    ikey.size = 4;
    ival.data = (void*)"";
    ival.size = 0;
    iwkv_opflags opflags = IWKV_NO_OVERWRITE;
    iwrc rc = iwkv_put( db, &ikey, &ival, opflags );
    assert( rc == 0 );
}
````

**Comment by adamansky** [comment]
Hi! Will check it

**Comment by doublex** [comment]
@adamansky 
Thanks a lot. Did I make a mistake?

**Comment by adamansky** [comment]
Not its my mistake =) 06ce001

**Comment by adamansky** [comment]
Released v1.3.30
```
iowow (1.3.30) testing; urgency=medium

  * Added check if  __unused is already defined in `iwth.c`
  * CRITICAL Comparsions keys fix #30
```

**Comment by doublex** [comment]
Thanks for your help!
